android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java

```diff
@@ -82,7 +94,18 @@ protected void onCreate(Bundle savedInstanceState) {
             showUserData(user);
         });
 
-        avatar.setOnClickListener(v -> showFileChooser());
+        permissionGranted = checkReadExternalPermission();
+        Observable<Object> picRequestObserver = Observable.create(emitter -> ActivityCompat.requestPermissions(EditProfileActivity.this, READ_STORAGE, REQUEST_CODE));
+        picRequestObserver.subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread());
```

Comment:
There is simply no use of Observable here at all

How does wrapping in observer help? Please follow up with a discussion, not a commit or else we'll be stuck in this loop for days on end

Reply:
Yeah. I too thought simply that it wont be of any use. I am not really able to get the part where we should use reactive

Reply:
@iamareebjamal Where exactly do we need reactive?

Reply:
See the pic @simarsingh24 posted above, you need to react to the permission grant or failure correctly

Read the documentation on how to handle the permission access

Reply:
https://user-images.githubusercontent.com/16715499/40821518-149e4baa-6585-11e8-819b-d0cbc55d4029.png

Also, follow up with what you are going to do so that we can understand and approve. Committing another solution after hours to be requested again is not a good way to iterate on solutions. We should have been on same page since starting the issue.